Detectors/TRD/workflow/src/TRDTrapSimulatorSpec.cxx

```diff
@@ -266,82 +279,74 @@ void TRDDPLTrapSimulatorTask::run(o2::framework::ProcessingContext& pc)
+
+  /*********
+   * iNPUTS
+   ********/
+
   auto inputDigits = pc.inputs().get<gsl::span<o2::trd::Digit>>("digitinput");
   std::vector<o2::trd::Digit> msgDigits(inputDigits.begin(), inputDigits.end());
   //  auto digits pc.outputs().make<std::vector<o2::trd::Digit>>(Output{"TRD", "TRKDIGITS", 0, Lifetime::Timeframe}, msgDigits.begin(), msgDigits.end());
   auto digitMCLabels = pc.inputs().get<o2::dataformats::ConstMCTruthContainer<o2::MCCompLabel>>("labelinput");
-
-  //  auto rawDataOut = pc.outputs().make<char>(Output{"TRD", "RAWDATA", 0, Lifetime::Timeframe}, 1000); //TODO number is just a place holder until we start using it.
-  o2::dataformats::MCTruthContainer<o2::MCCompLabel> trackletMCLabels;
-
   // the returned object is read-only as it refers directly to the underlying raw input data
   // need to make a copy because the object might be changed in fixTriggerRecords
   auto inputTriggerRecords = pc.inputs().get<gsl::span<o2::trd::TriggerRecord>>("triggerrecords");
+
+  /* *****
+   * setup data objects
+   * *****/
+
   // trigger records to index the 64bit tracklets.yy
   std::vector<o2::trd::TriggerRecord> triggerRecords(inputTriggerRecords.begin(), inputTriggerRecords.end());
+  std::vector<o2::trd::TriggerRecord> trackletTriggerRecords = triggerRecords; // copy over the whole thing but we only really want the bunch crossing info.
+  std::vector<o2::trd::TriggerRecord> rawTriggerRecords = triggerRecords;      // as we have the option of having tracklets and/or raw data, we need both triggerrecords.
+                                                                               // of course we dont *actually* need it we could simply walk through all the raw data header to header.
+  //  auto rawDataOut = pc.outputs().make<char>(Output{"TRD", "RAWDATA", 0, Lifetime::Timeframe}, 1000); //TODO number is just a place holder until we start using it.
+  o2::dataformats::MCTruthContainer<o2::MCCompLabel> trackletMCLabels;
+  //index of digits, TODO refactor to a digitindex class.
+  std::vector<unsigned int> msgDigitsIndex;
+  //set up structures to hold the returning tracklets.
+  std::vector<Tracklet64> trapTracklets; //vector to store the retrieved tracklets from an trapsim object
+  std::vector<Tracklet64> trapTrackletsAccum;
+  std::vector<uint32_t> rawdata;
   // trigger records to index the "raw" data
-
   uint64_t currentTriggerRecord = 0;
-
   for (auto& trig : triggerRecords) {
     if (mDumpTriggerRecords) {
-      LOG(info) << "Trigger Record ; " << trig.getFirstEntry() << " --> " << trig.getNumberOfObjects();
+      LOG(info) << "Trigger Record : " << trig.getFirstEntry() << " --> " << trig.getNumberOfObjects();
     } else {
-      LOG(debug) << "Trigger Record ; " << trig.getFirstEntry() << " --> " << trig.getNumberOfObjects();
+      LOG(debug) << "Trigger Record : " << trig.getFirstEntry() << " --> " << trig.getNumberOfObjects();
     }
   }
   // fix incoming trigger records if requested.
   if (mFixTriggerRecords) {
```

Comment:
Is the fix for trigger records still needed?

Reply:
No

Reply:
removed